controllers/topology/reconcile_state.go

```diff
@@ -100,7 +100,14 @@ func (r *ClusterReconciler) reconcileControlPlane(ctx context.Context, s *scope.
 
 	// Create or update the ControlPlaneObject for the ControlPlaneState.
 	ctx, _ = tlog.LoggerFrom(ctx).WithObject(s.Desired.ControlPlane.Object).Into(ctx)
-	if err := r.reconcileReferencedObject(ctx, s.Current.ControlPlane.Object, s.Desired.ControlPlane.Object); err != nil {
+	if err := r.reconcileReferencedObject(ctx, s.Current.ControlPlane.Object, s.Desired.ControlPlane.Object, mergepatch.AuthoritativePaths{
+		// Note: we want to be authoritative WRT machine's metadata labels and annotations.
+		// This has the nice benefit that it greatly simplify the UX around ControlPlaneClass.Metadata and
+		// ControlPlaneTopology.Metadata, given that changes are reflected into generated objects without
+		// accounting for instance specific changes like we do for other maps into spec.
+		// Note: Patch helpers already reconcile only metadata labels and annotations, so no need to be more specific here.
```

Comment:
But the mergepatch helper is only filtering top-level `.metadata.{labels,annotations}` not `.spec.machineTemplate.metadata.{labels,annotations}`, right?

(I know it's not a problem in KCP as KCP metadata only has labels/annotations, not sure if other control plane providers could diverge)

P.S. I'm probably misinterpreting something :)

Reply:
added a comment specific for nested metadata

Reply:
lgtm +/- the commit is pushed + squash :)